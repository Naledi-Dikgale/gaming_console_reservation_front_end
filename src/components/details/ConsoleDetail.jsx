import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
function ConsoleDetail() {
  const [theConsole, SettheConsole] = useState(null)
  const { id } = useParams();
  const consoles = useSelector((state) => state.consoles.consoles);

  useEffect(() => {
    consoles.forEach(element => {
      if (element.id === parseInt(id)) {
        SettheConsole(element);
      }
    });
    console.log(consoles);
  }, [id, consoles])

  return (
    <div className="flex flex-col items-center justify-center h-full">
      {theConsole ? (
        <div className="flex w-full justify-between items-center">
          <div className="w-3/5">
            <img src={theConsole.photo} alt={theConsole.name} className="w-full h-auto" />
          </div>
          <div className="flex flex-col w-2/5 px-6">
            <div>
              <p className="text-lg font-bold p-1">{theConsole.name}</p>
              <p className=" text-xs p-1">{theConsole.description}</p>
              <p className="text-md bg-secondary p-1">Purchase Price: {theConsole.purchase_price} $</p>
              <p className="text-md p-1">Rental Price: {theConsole.rental_price} $</p>
            </div>
            <div className="flex flex-col w-full justify-center items-center">
              <button
                className="w-2/5 bg-secondary text-primary rounded-lg p-2 mt-3">Reserve</button>
            </div>
          </div>
        </div>
      ) : (
        <div className="text-center">Console not found</div>
      )}
    </div>
  );
}


export default ConsoleDetail;
