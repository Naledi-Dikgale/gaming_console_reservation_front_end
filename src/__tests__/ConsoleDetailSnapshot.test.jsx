import { render } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { describe, test, expect } from 'vitest'; // Importing from vitest
import ConsoleDetail from '../components/details/ConsoleDetail';

const mockStore = configureStore([]);

describe('ConsoleDetail', () => {
  test('matches snapshot', () => {
    const consoleData = {
      id: 1,
      name: 'SEGA',
      purchase_price: 10,
      rental_price: 4,
      description: 'A good one',
      photo: 'linktoImage.png',
    };

    const initialState = {
      consoles: {
        consoles: [consoleData],
      },
    };

    const store = mockStore(initialState);

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/console/1']}>
          <Routes>
            <Route path="/console/:id" element={<ConsoleDetail />} />
          </Routes>
        </MemoryRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
