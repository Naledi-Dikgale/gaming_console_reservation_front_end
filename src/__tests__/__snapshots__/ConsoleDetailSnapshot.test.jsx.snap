// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConsoleDetail > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center h-full"
  >
    <div
      class="flex w-full justify-between items-center"
    >
      <div
        class="w-3/5"
      >
        <img
          alt="SEGA"
          class="w-full h-auto"
          src="linktoImage.png"
        />
      </div>
      <div
        class="flex flex-col w-2/5 px-6"
      >
        <div>
          <p
            class="text-lg font-bold p-1"
          >
            SEGA
          </p>
          <p
            class=" text-xs p-1"
          >
            A good one
          </p>
          <p
            class="text-md bg-secondary p-1"
          >
            Purchase Price: 
            10
             $
          </p>
          <p
            class="text-md p-1"
          >
            Rental Price: 
            4
             $
          </p>
        </div>
        <div
          class="flex flex-col w-full justify-center items-center"
        >
          <a
            href="/reserve/1"
          >
            <button
              class="w-full bg-secondary text-primary rounded-lg p-2 mt-3"
              type="button"
            >
              Reserve
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
